File: CS/CustomImageEditor/DragImageEdit/DragImageEditViewInfo.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Controls;

namespace CustomImageEditor {
    public class DragImageEditViewInfo : BaseEditViewInfo {
        Image image;
        int _VerticalOffset, _HorizontOffset;
        public Image Image {
            get { return image; }
            set {
                image = value;
            }
        }
        public int VerticalOffset {
            get { return _VerticalOffset;}
            set{_VerticalOffset = value;}
        }
        public int HorizontOffset {
            get {return _HorizontOffset;}
            set{_HorizontOffset = value;}
        }
   
        protected override void OnEditValueChanged() {
            if(!(EditValue is EditValueInstance)) return;
            if((EditValue != null) && (EditValue as EditValueInstance).Image != null)
                this.image = ByteImageConverter.FromByteArray(ByteImageConverter.ToByteArray((EditValue as EditValueInstance).Image));
            Point offset = new Point(0, 0);
            if((EditValue as EditValueInstance) != null) { offset = (EditValue as EditValueInstance).Offset; }
            if((EditValue != null) && (offset != null)) {
                this.HorizontOffset = offset.X;
                this.VerticalOffset = offset.Y;
            }
        }

        protected override void Assign(BaseControlViewInfo info) {
            base.Assign(info);
            DragImageEditViewInfo be = info as DragImageEditViewInfo;
            if(be == null)
                return;
            this.Image = be.Image;
            this._HorizontOffset = be._HorizontOffset;
            this._VerticalOffset = be._VerticalOffset;
        }

        public override void Reset() {
            if(EditValue != null) {
                _HorizontOffset = 0;
                _VerticalOffset = 0;
            }
            base.Reset();
            this.image = null;
           
        }
        public new DragImagEditRepositoryItem Item { get { return base.Item as DragImagEditRepositoryItem; } }
        public DragImageEditViewInfo(RepositoryItem item)
            : base(item) {
        }
    }
}

File: VB/CustomImageEditor/DragImageEdit/DragImageEditViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Controls

Namespace CustomImageEditor
	Public Class DragImageEditViewInfo
		Inherits BaseEditViewInfo
		Private image_Renamed As Image
		Private _VerticalOffset, _HorizontOffset As Integer
		Public Property Image() As Image
			Get
				Return image_Renamed
			End Get
			Set(ByVal value As Image)
				image_Renamed = value
			End Set
		End Property
		Public Property VerticalOffset() As Integer
			Get
				Return _VerticalOffset
			End Get
			Set(ByVal value As Integer)
				_VerticalOffset = value
			End Set
		End Property
		Public Property HorizontOffset() As Integer
			Get
				Return _HorizontOffset
			End Get
			Set(ByVal value As Integer)
				_HorizontOffset = value
			End Set
		End Property

		Protected Overrides Sub OnEditValueChanged()
			If Not(TypeOf EditValue Is EditValueInstance) Then
				Return
			End If
			If (EditValue IsNot Nothing) AndAlso (TryCast(EditValue, EditValueInstance)).Image IsNot Nothing Then
				Me.image_Renamed = ByteImageConverter.FromByteArray(ByteImageConverter.ToByteArray((TryCast(EditValue, EditValueInstance)).Image))
			End If
			Dim offset As New Point(0, 0)
			If (TryCast(EditValue, EditValueInstance)) IsNot Nothing Then
				offset = (TryCast(EditValue, EditValueInstance)).Offset
			End If
            If (EditValue IsNot Nothing) AndAlso (Not IsNothing(offset)) Then
                Me.HorizontOffset = offset.X
                Me.VerticalOffset = offset.Y
            End If
		End Sub

		Protected Overrides Sub Assign(ByVal info As BaseControlViewInfo)
			MyBase.Assign(info)
			Dim be As DragImageEditViewInfo = TryCast(info, DragImageEditViewInfo)
			If be Is Nothing Then
				Return
			End If
			Me.Image = be.Image
			Me._HorizontOffset = be._HorizontOffset
			Me._VerticalOffset = be._VerticalOffset
		End Sub

		Public Overrides Sub Reset()
			If EditValue IsNot Nothing Then
				_HorizontOffset = 0
				_VerticalOffset = 0
			End If
			MyBase.Reset()
			Me.image_Renamed = Nothing

		End Sub
		Public Shadows ReadOnly Property Item() As DragImagEditRepositoryItem
			Get
				Return TryCast(MyBase.Item, DragImagEditRepositoryItem)
			End Get
		End Property
		Public Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)
		End Sub
	End Class
End Namespace
